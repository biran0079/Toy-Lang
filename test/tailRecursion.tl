fun f(n){
  if(n) f(n-1);
}
print(f(50000));

fun sum2(n, res){
  if(!n)return res;
  return sum2(n-1, res+n);
}
print(sum2(50000,0));
