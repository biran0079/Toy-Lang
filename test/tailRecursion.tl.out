none
1250025000
